tiledb/sm/query/test/unit_query_condition.cc

```diff
@@ -1674,45 +1685,46 @@ void test_apply(const Datatype type, bool var_size, bool nullable) {
 }
 
 TEST_CASE("QueryCondition: Test apply", "[QueryCondition][apply]") {
-  test_apply<int8_t>(Datatype::INT8);
-  test_apply<uint8_t>(Datatype::UINT8);
-  test_apply<int16_t>(Datatype::INT16);
-  test_apply<uint16_t>(Datatype::UINT16);
-  test_apply<int32_t>(Datatype::INT32);
-  test_apply<uint32_t>(Datatype::UINT32);
-  test_apply<int64_t>(Datatype::INT64);
-  test_apply<uint64_t>(Datatype::UINT64);
-  test_apply<float>(Datatype::FLOAT32);
-  test_apply<double>(Datatype::FLOAT64);
-  test_apply<char>(Datatype::CHAR);
-  test_apply<int64_t>(Datatype::DATETIME_YEAR);
-  test_apply<int64_t>(Datatype::DATETIME_MONTH);
-  test_apply<int64_t>(Datatype::DATETIME_WEEK);
-  test_apply<int64_t>(Datatype::DATETIME_DAY);
-  test_apply<int64_t>(Datatype::DATETIME_HR);
-  test_apply<int64_t>(Datatype::DATETIME_MIN);
-  test_apply<int64_t>(Datatype::DATETIME_SEC);
-  test_apply<int64_t>(Datatype::DATETIME_MS);
-  test_apply<int64_t>(Datatype::DATETIME_US);
-  test_apply<int64_t>(Datatype::DATETIME_NS);
-  test_apply<int64_t>(Datatype::DATETIME_PS);
-  test_apply<int64_t>(Datatype::DATETIME_FS);
-  test_apply<int64_t>(Datatype::DATETIME_AS);
-  test_apply<int64_t>(Datatype::TIME_HR);
-  test_apply<int64_t>(Datatype::TIME_MIN);
-  test_apply<int64_t>(Datatype::TIME_SEC);
-  test_apply<int64_t>(Datatype::TIME_MS);
-  test_apply<int64_t>(Datatype::TIME_US);
-  test_apply<int64_t>(Datatype::TIME_NS);
-  test_apply<int64_t>(Datatype::TIME_PS);
-  test_apply<int64_t>(Datatype::TIME_FS);
-  test_apply<int64_t>(Datatype::TIME_AS);
-  test_apply<char*>(Datatype::STRING_ASCII);
-  test_apply<char*>(Datatype::STRING_ASCII, true);
-  test_apply<char*>(Datatype::STRING_ASCII, false, true);
-  test_apply<char*>(Datatype::STRING_UTF8);
-  test_apply<char*>(Datatype::STRING_UTF8, true);
-  test_apply<char*>(Datatype::STRING_UTF8, false, true);
+  auto tracker = tiledb::test::create_test_memory_tracker();
```

Comment:
You can undo this and just use `tiledb::test::get_test_memory_tracker()` now. I swapped the create to a process global instance in the test suite.